### DISCUSSION TITLE
Indicate that a troop has already performed its action (move, attack, pass its turn, etc.) for the current turn in the troop information window

### DISCUSSION THREAD

**Discussion by LeHerosInconnu** [ihhub/fheroes2#2532] [Ideas]
I will make a suggestion later. :)

**Edit.**

For now, a simple colored icon could do the trick:

- Green icon, the troop has not yet performed its action for the current tour.
- Red icon, the troop has already performed its action for the current turn.
- An additional yellow icon with green or red square inside indicates if the troop has already performed or not its retaliation for the current turn (this can be another issue).

More elaborate icons can be added later.

In fheroes2:

![Wrong creature speed information 05 01](https://user-images.githubusercontent.com/43583994/103483822-29913400-4dea-11eb-96e7-ac6ffc580862.png)

![Wrong creature speed information 05 02](https://user-images.githubusercontent.com/43583994/103483833-3dd53100-4dea-11eb-945b-2bc870f09ffb.png)

![Wrong creature speed information 05 04](https://user-images.githubusercontent.com/43583994/103484046-01a2d000-4dec-11eb-825f-f7074b08d25f.png)

![Wrong creature speed information 05 05](https://user-images.githubusercontent.com/43583994/103484054-0ff0ec00-4dec-11eb-9c30-d85ecff80641.png)

Screens are for explanation only. :)
The position of the icons can be changed.

**Comment by Branikolog** [comment]
Hi, @LeHerosInconnu !
Adding action indicator is a excellent idea. But I can accept it only if it will be pretty and intelligible. :D
Players should clearly understand what does each type of this indicator mean. I think for such implementation we have to use help of good artist... 
Could you tell me please, if we have an opened issue for a case, when "speed number" of a creature, that performed an action turns to "0"?
If not, could you create, please? If we have numeral description for creatures' speed, it should always be the same in creature info window....

**Reply by LeHerosInconnu** [reply]
@Branikolog.

> Hi, @LeHerosInconnu !
> Adding action indicator is a excellent idea. But I can accept it only if it will be pretty and intelligible. :D
> Players should clearly understand what does each type of this indicator mean. I think for such implementation we have to use help of good artist...

As I wrote, the screens are there for explanation.
Beautiful icons should be used. :)

> Could you tell me please, if we have an opened issue for a case, when "speed number" of a creature, that performed an action turns to "0"?
> If not, could you create, please? If we have numeral description for creatures' speed, it should always be the same in creature info window....

This has already been fixed here : https://github.com/ihhub/fheroes2/issues/1544.

**Reply by Branikolog** [reply]
> As I wrote, the screens are there for explanation.
> Beautiful icons should be used. :)

Until we have an artist... (or you proposed an idea using current UI elements) We should at least discuss the place for this indicator...

Between, what do you think, if it would be not an indicator, but some sort of inscription about the current state? (My English is quite poor, so I cannot offer nice words... Maybe something like "Ready to act!", "Waiting for an action.", "Already took an action this turn.", "Going to act now..." or "Blinded: cannot perform any action.") 
As the right side of info window is quite overfull, we can place this inscription right under the title of a creature in the left part of this window. (It could be in a different font color, comparing to yellow title.)
Left part of this window is quite empty, And I think, a small single-line description will look rather suitable to the game so rich with text descriptions and stories.

P.S.

> This has already been fixed here : #1544.

I start forgetting not only the issues created, but also issues been already fixed.  o_O

**Reply by LeHerosInconnu** [reply]
@Branikolog.

> Between, what do you think, if it would be not an indicator, but some sort of inscription about the current state? (My English is quite poor, so I cannot offer nice words... Maybe something like "Ready to act!", "Waiting for an action.", "Already took an action this turn.", "Going to act now..." or "Blinded: cannot perform any action.")
> As the right side of info window is quite overfull, we can place this inscription right under the title of a creature in the left part of this window. (It could be in a different font color, comparing to yellow title.)
> Left part of this window is quite empty, And I think, a small single-line description will look rather suitable to the game so rich with text descriptions and stories.

I have planed to propose to add a description about the creature's abilities and immunities (special abilities, immunity to spells, etc.) at the bottom left of this window between the creature's sprite and the creature counter.
It will certainly be necessary to move the position of the creature's sprite slightly upwards.
So I'm not very enthusiastic about placing more text in this window.

An icon is also more discreet and is quickly recognizable.
And with an icon there is no need to translate the text into several languages. :)

> P.S.
> 
> > This has already been fixed here : #1544.
> 
> I start forgetting not only the issues created, but also issues been already fixed. o_O

:rofl:

**Reply by Branikolog** [reply]
Well, if we make icons, we'd have to create graphic manual for players, as several icons are much less recognizable, especially for new players...
Anyway... Looking at your presentation in these screenshots, I find the place you've put icons... The right side of this window is full of items, and placing new icons in the lowest row  makes window look not good...

**Reply by LeHerosInconnu** [reply]
@Branikolog.

> Well, if we make icons, we'd have to create graphic manual for players, as several icons are much less recognizable, especially for new players...

Not really a need for a manual.
As it is already the case in the game, a right click on the icon should indicate what it represents.

> Anyway... Looking at your presentation in these screenshots, I find the place you've put icons... The right side of this window is full of items, and placing new icons in the lowest row makes window look not good...

As I wrote:
> Screens are for explanation only. :)
> The position of the icons can be changed.

**Reply by Branikolog** [reply]
> The position of the icons can be changed.

So, we're here to discuss.)

Sorry for only criticizing your ideas without proposing my... 
You have so many ideas, @LeHerosInconnu, that I barely have enough enthusiasm to read them, think about them and analyzing what I'm going to like and what could be improved. :D 

> As it is already the case in the game, a right click on the icon should indicate what it represents.

Yeah, I had the same thoughts.
But the thing is that people hardly ever use left click to open a static info window. They more often use simple right click, where you cannot check, what is this status icon means...

**Reply by LeHerosInconnu** [reply]
@Branikolog.

> > The position of the icons can be changed.
> 
> So, we're here to discuss.)

As the icons concern the attack and retaliation of the troop, positioning the icons in the extension of the line concerning the base of the attack range seems appropriate:

![Attack retaliation icons fh2 01](https://user-images.githubusercontent.com/43583994/108282523-0fd16380-7182-11eb-99fa-8b04ac13696c.png)

> Sorry for only criticizing your ideas without proposing my...
> You have so many ideas, @LeHerosInconnu, that I barely have enough enthusiasm to read them, think about them and analyzing what I'm going to like and what could be improved. :D

Stay tuned. :D

> > As it is already the case in the game, a right click on the icon should indicate what it represents.
> 
> Yeah, I had the same thoughts.
> But the thing is that people hardly ever use left click to open a static info window. They more often use simple right click, where you cannot check, what is this status icon means...

This is a "problem" of the interface.

Modern interfaces have tooltips everywhere (moving the mouse without clicking on elements (a button, a graphic, etc.) makes a "mini window" appear), but I find it annoying and invasive (because the window appears even when it is not a "voluntary" action).

But the information is accessible if the player uses the interface, and once he knows what the icons represent, it's rather easy to understand.

**Reply by Branikolog** [reply]
When I was talking about graphic manual, I was thinking about some kind of illustrated "readme" file, where player can get acquainted with new mechanics and features, like new icons, splitting combinations etc. 
I really miss nice guide for H3 for example, that show all the improvements...

**Reply by LeHerosInconnu** [reply]
@Branikolog.

> When I was talking about graphic manual, I was thinking about some kind of illustrated "readme" file, where player can get acquainted with new mechanics and features, like new icons, splitting combinations etc.
> I really miss nice guide for H3 for example, that show all the improvements...

Glad you volunteered to write this manual. :D

**Reply by Branikolog** [reply]
Well... 
Cannot tell, that I'm glad either. ^_^
Anyway, close to 1.0 version we have to introduce such readme.)

**Comment by LeHerosInconnu** [comment]
This is planned in my mod: https://github.com/ihhub/fheroes2/discussions/10101.
I close the discussion.
